shell/common/shell.cc

```diff
@@ -67,58 +67,57 @@ std::unique_ptr<Shell> Shell::CreateShellOnPlatformThread(
   // first because it has state that the other subsystems depend on. It must
   // first be booted and the necessary references obtained to initialize the
   // other subsystems.
-  fml::AutoResetWaitableEvent io_latch;
-  std::unique_ptr<ShellIOManager> io_manager;
+  std::promise<std::unique_ptr<ShellIOManager>> io_manager_promise;
+  auto io_manager = io_manager_promise.get_future();
+  std::promise<fml::WeakPtr<ShellIOManager>> weak_io_manager_promise;
+  auto weak_io_manager = weak_io_manager_promise.get_future();
   auto io_task_runner = shell->GetTaskRunners().GetIOTaskRunner();
   fml::TaskRunner::RunNowOrPostTask(
       io_task_runner,
-      [&io_latch,       //
-       &io_manager,     //
-       &platform_view,  //
-       io_task_runner   //
+      [&io_manager_promise,       //
+       &weak_io_manager_promise,  //
+       &platform_view,            //
+       io_task_runner             //
   ]() {
         TRACE_EVENT0("flutter", "ShellSetupIOSubsystem");
-        io_manager = std::make_unique<ShellIOManager>(
+        auto io_manager = std::make_unique<ShellIOManager>(
             platform_view->CreateResourceContext(), io_task_runner);
-        io_latch.Signal();
+        weak_io_manager_promise.set_value(io_manager->GetWeakPtr());
+        io_manager_promise.set_value(std::move(io_manager));
       });
-  io_latch.Wait();
 
   // Create the rasterizer on the GPU thread.
-  fml::AutoResetWaitableEvent gpu_latch;
-  std::unique_ptr<Rasterizer> rasterizer;
+  std::promise<std::unique_ptr<Rasterizer>> rasterizer_promise;
```

Comment:
Please move this before the platform view creation call. You don't have to wait for the platform view to be created before this can be kicked off as there is no dependency. This can be kicked off eagerly so that platform view creation can be concurrent with rasterizer creation as well.

Reply:
done